src/client/systems/NetworkingClient.cpp

```diff
@@ -260,9 +260,34 @@ void NetworkingClient::unreliableSend() {
     }
 }
 
+void NetworkingClient::sendNatPunch() {
+    sf::Packet packet;
+    sf::Uint32 unreliable_input_cmd = UnreliableCommandType::nat_punch;
+    UnreliableCommand unreliable_command = {(sf::Uint32)m_clientId_ta, unreliable_input_cmd,
+                                            m_tick_ta};
+
+    std::cout << "we sending a nat punch" << std::endl;
+    if (packet << unreliable_command) {
+        std::lock_guard<std::mutex> m_udpSocket_guard(m_udpSocket_lock);
+        sf::Socket::Status status = sf::Socket::Partial;
+        while (status == sf::Socket::Partial) {
+            status = m_udpSocket_t->send(packet, SERVER_IP, m_serverUdpPort);
+        }
+    } else {
+        std::cout << "Failed to form packet" << std::endl;
+    }
+}
+
 void NetworkingClient::sendUnreliableInput() {
     std::lock_guard<std::mutex> m_unreliableInputs_guard(m_unreliableInputs_lock);
     if (m_unreliableInputs_t.empty()) {
+        srand(time(0));
+        bool send_nat_punch = (rand() % 10 == 0);
```

Comment:
we basically dont want to constantly send nat punches, but how do we decide when to send it? basically we should send it if we haven't sent an input command recently. this was my quick an dirty implementation which didnt involve keeping a timestamp for last sent input command, but I think Ill just switch to the timestamp solution